workflow-manager/src/workflow_update.rs

```diff
@@ -0,0 +1,284 @@
+            return Err(ManagerError::IdIsZero);
+        }
+
+        let mut config = neutron_connector.get_workflow_config(self.id).await?;
+
+        let mut instructions: VecDeque<CosmosMsg> = VecDeque::new();
+        let mut new_authorizations: Vec<AuthorizationInfo> = vec![];
+
+        if let Some(new_owner) = self.owner.clone() {
+            config.owner = new_owner.clone();
+
+            // Create instruction to change owner
+            instructions.push_back(
+                WasmMsg::Execute {
+                    contract_addr: config.authorization_data.authorization_addr.clone(),
+                    msg: to_json_binary(&cw_ownable::Action::TransferOwnership {
+                        new_owner,
+                        expiry: None,
+                    })
+                    .context("Failed binary parsing TransferOwnership")?,
+                    funds: vec![],
+                }
+                .into(),
+            );
+        }
+
+        for (id, service_update) in self.services.iter() {
+            // Verify that the service id exists in the config and get it
+            let service = config
+                .services
+                .get(id)
+                .context(ManagerError::ServiceIdIsMissing(*id).to_string())?;
+
+            // Add authorization to update the service
+            let label = format!("update_service_{}_{}", service.name, id);
+            let service_domain = if service.domain == neutron_domain {
+                valence_authorization_utils::domain::Domain::Main
+            } else {
+                valence_authorization_utils::domain::Domain::External(service.domain.to_string())
+            };
+            let actions_config = AtomicActionsConfigBuilder::new()
+                .with_action(
+                    AtomicActionBuilder::new()
+                        .with_domain(service_domain)
+                        .with_contract_address(ServiceAccountType::Addr(
+                            service.addr.clone().unwrap(),
+                        ))
+                        .with_message_details(MessageDetails {
+                            message_type: MessageType::CosmwasmExecuteMsg,
+                            message: Message {
+                                name: "update_config".to_string(),
+                                params_restrictions: None,
+                            },
+                        })
+                        .build(),
+                )
+                .build();
+            let authorization_builder = AuthorizationBuilder::new()
+                .with_label(&label)
+                .with_mode(AuthorizationModeInfo::Permissioned(
+                    valence_authorization_utils::authorization::PermissionTypeInfo::WithoutCallLimit(vec![config.owner.clone()]),
+                ))
+                .with_priority(Priority::High)
+                .with_actions_config(actions_config);
+
+            new_authorizations.push(authorization_builder.build());
```

Comment:
Should you check if it doesn't exist yet before adding it ?

Reply:
Good point